frontend/public/components/monitoring/dashboards/index.tsx

```diff
@@ -105,7 +112,27 @@ const VariableDropdown: React.FC<VariableDropdownProps> = ({
 }) => {
   const { t } = useTranslation();
 
-  const [isOpen, toggleIsOpen, , setClosed] = useBoolean(false);
+  const [isOpen, , open, close] = useBoolean(false);
+  const [filterText, setFilterText] = React.useState<string>();
+
+  const onSelect = (e, value: string): void => {
+    onChange(value);
+    close();
+    setFilterText(undefined);
+  };
+
+  const onToggle = (isExpanded: boolean) => {
+    if (isExpanded) {
+      open();
+    } else {
+      close();
+      setFilterText(undefined);
+    }
+  };
+
+  const filteredItems = filterText
+    ? _.pickBy(items, (v) => v.toLowerCase().includes(filterText))
```

Comment:
Worth a comment explaining the `filterText` is lower-cased before it's saved in state.